### DISCUSSION TITLE
in remember for login logic, the login isn't skipped

### DISCUSSION THREAD

**Discussion by happyjyj** [ory/hydra#3087] [Q&A] [answered]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/hydra/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/hydra/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

1. The first-time login with remember succeeds.
2. Call authorize endpoint with prompt=none in the same browser with cookies, the cookies include the following 
<img width="1492" alt="image" src="https://user-images.githubusercontent.com/99728256/164954621-50d250bb-e7c4-4489-b757-092bce676ced.png">
However, it always returns the login URL with login challenge. 

I checked the source code, it seems to return the login URL with login challenge every time.
`func (s *DefaultStrategy) forwardAuthenticationRequest(w http.ResponseWriter, r *http.Request, ar fosite.AuthorizeRequester, subject string, authenticatedAt time.Time, session *LoginSession) error {
	if (subject != "" && authenticatedAt.IsZero()) || (subject == "" && !authenticatedAt.IsZero()) {
		return errorsx.WithStack(fosite.ErrServerError.WithHint("Consent strategy returned a non-empty subject with an empty auth date, or an empty subject with a non-empty auth date."))
	}

	skip := false
	if subject != "" {
		skip = true
	}

	// Let'id validate that prompt is actually not "none" if we can't skip authentication
	prompt := stringsx.Splitx(ar.GetRequestForm().Get("prompt"), " ")
	if stringslice.Has(prompt, "none") && !skip {
		return errorsx.WithStack(fosite.ErrLoginRequired.WithHint(`Prompt 'none' was requested, but no existing login session was found.`))
	}

	// Set up csrf/challenge/verifier values
	verifier := strings.Replace(uuid.New(), "-", "", -1)
	challenge := strings.Replace(uuid.New(), "-", "", -1)
	csrf := strings.Replace(uuid.New(), "-", "", -1)

	// Generate the request URL
	iu := s.c.OAuth2AuthURL()
	iu.RawQuery = r.URL.RawQuery

	var idTokenHintClaims jwtgo.MapClaims
	if idTokenHint := ar.GetRequestForm().Get("id_token_hint"); len(idTokenHint) > 0 {
		claims, err := s.getIDTokenHintClaims(r.Context(), idTokenHint)
		if err != nil {
			return err
		}

		idTokenHintClaims = claims
	}

	sessionID := uuid.New()
	if session != nil {
		sessionID = session.ID
	} else {
		// Create a stub session so that we can later update it.
		if err := s.r.ConsentManager().CreateLoginSession(r.Context(), &LoginSession{ID: sessionID}); err != nil {
			return err
		}
	}

	// Set the session
	if err := s.r.ConsentManager().CreateLoginRequest(
		r.Context(),
		&LoginRequest{
			ID:                challenge,
			Verifier:          verifier,
			CSRF:              csrf,
			Skip:              skip,
			RequestedScope:    []string(ar.GetRequestedScopes()),
			RequestedAudience: []string(ar.GetRequestedAudience()),
			Subject:           subject,
			Client:            sanitizeClientFromRequest(ar),
			RequestURL:        iu.String(),
			AuthenticatedAt:   sqlxx.NullTime(authenticatedAt),
			RequestedAt:       time.Now().Truncate(time.Second).UTC(),
			SessionID:         sqlxx.NullString(sessionID),
			OpenIDConnectContext: &OpenIDConnectContext{
				IDTokenHintClaims: idTokenHintClaims,
				ACRValues:         stringsx.Splitx(ar.GetRequestForm().Get("acr_values"), " "),
				UILocales:         stringsx.Splitx(ar.GetRequestForm().Get("ui_locales"), " "),
				Display:           ar.GetRequestForm().Get("display"),
				LoginHint:         ar.GetRequestForm().Get("login_hint"),
			},
		},
	); err != nil {
		return errorsx.WithStack(err)
	}

	if err := createCsrfSession(w, r, s.r.CookieStore(), cookieAuthenticationCSRFName, csrf, s.c.TLS(config.PublicInterface).Enabled(), s.c.CookieSameSiteMode(), s.c.CookieSameSiteLegacyWorkaround()); err != nil {
		return errorsx.WithStack(err)
	}

	http.Redirect(w, r, urlx.SetQuery(s.c.LoginURL(), url.Values{"login_challenge": {challenge}}).String(), http.StatusFound)

	// generate the verifier
	return errorsx.WithStack(ErrAbortOAuth2Request)
}`

### Reproducing the bug

1. login and consent succeed. the login is set to remember.
2. Call authorize endpoint with prompt=none in the same browser with cookies.

### Version

v1.10.3

### On which operating system are you observing this issue?

Linux

### In which environment are you deploying?

Kubernetes

**Comment by happyjyj** [comment]
Did I miss any configuration to activate this feature?

**Comment by aeneasr** [comment, marked as answer]
Since this works in all automated tests and deployments across the board, it's most likely a configuration issue (e.g. domain issue or HTTPS issue). Someone from dev rel / community can for sure answer this :)

**Reply by happyjyj** [reply]
Thanks. Here are the environment variables that I used to run hydra.
`
DSN=mysql://xxxxxxxxxxx;
URLS_CONSENT=https://xxxxxxx/consent;
URLS_LOGIN=xxxxxxxxxxxxx;
SECRETS_SYSTEM=xxxxxxx;
STRATEGIES_ACCESS_TOKEN=jwt;
WEBFINGER_OIDC_DISCOVERY_AUTH_URL=https://xxxxxxx/authorize;
WEBFINGER_OIDC_DISCOVERY_TOKEN_URL=https://xxxxxxx/token;
WEBFINGER_OIDC_DISCOVERY_USERINFO_URL=https://xxxxxxx/userinfo;
WEBFINGER_OIDC_DISCOVERY_JWKS_URL=https://xxxxxxx/.well-known/jwks.json;
WEBFINGER_OIDC_DISCOVERY_SUPPORTED_SCOPE=xxxxxxxxxxxxxxxx;
STRATEGIES_SCOPE=DEPRECATED_HIERARCHICAL_SCOPE_STRATEGY;
WEBFINGER_OIDC_DISCOVERY_CLIENT_REGISTRATION_URL=https://xxxxxxx/register;
WEBFINGER_OIDC_DISCOVERY_SUPPORTED_CLAIMS=xxxxxxx;
OAUTH2_ALLOWED_TOP_LEVEL_CLAIMS="client_id";
URLS_SELF_ISSUER=https://xxxxxxx;
OAUTH2_Revocation_URL=https://xxxxxxx;
LOG_LEVEL=debug;
SERVE_ADMIN_TLS_ENABLED=true
`

Who can help to figure out what configuration I missed?

**Reply by vinckr** [reply]
Hello @happyjyj, 
Could you please share a reproducible repository with shell commands (e.g. hydra clients create, curl ...) that reproduces this issue?

**Reply by happyjyj** [reply]
Hi, thank you for your reply. I found the problem, I missed the GetLoginRequest step in my design. I am trying to verify it now.

**Comment by happyjyj** [comment]
@vinckr  Do you have any best practice design of SSO based on hydra? Does that include the logout flow?

**Reply by vinckr** [reply]
You can check out the community projects: https://www.ory.sh/docs/ecosystem/community
specifically this one: https://blog.px.dev/open-source-auth/#integrated-kratoshydra-design 
It will be easier once the Hydra/Kratos integration is more fleshed out, bear with us 🙏 
What are you using as authentication solution (e.g. Ory Kratos)?
